napari/components/_viewer/view/main.py

```diff
@@ -214,20 +218,44 @@ def on_mouse_release(self, event):
     def on_key_press(self, event):
         """Called whenever key pressed in canvas.
         """
-        if (
-            event.text in self.viewer.key_bindings
-            and not event.native.isAutoRepeat()
-        ):
-            self.viewer.key_bindings[event.text](self.viewer)
+        if event.native.isAutoRepeat() or event.key is None:
             return
 
+        comb = components_to_comb(event.key.name, event.modifiers)
+
         layer = self.viewer.active_layer
+
+        # TODO: remove me once keybinding system converted
```

Comment:
What does this mean? Meaning, reworking all existing keybindings to use the new system?

Incidentally, how does this system interact with the Qt menubar system etc?

Reply:
Precisely.

Getting this to work with Qt's built-in shortcuts will be tricky if we want to do it natively. Otherwise, it would be pretty simple to just do something like `Viewer.bind_key('Control-W', lambda v: v.close())`